modules/doctor/queue.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package doctor
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/nosql"
+	"code.gitea.io/gitea/modules/queue"
+	"code.gitea.io/gitea/modules/setting"
+	"gitea.com/lunny/levelqueue"
+)
+
+var uniqueQueueNames = []string{
+	"code_indexer",
+	"repo_stats_update",
+	"mirror",
+	"pr_patch_checker",
+	"repo-archive",
+}
+
+var queueNames = []string{
+	"issue_indexer",
+	"notification-service",
+	"mail",
+	"push_update",
+	"task",
+}
+
+var levelqueueTypes = []string{
+	string(queue.PersistableChannelQueueType),
+	string(queue.PersistableChannelUniqueQueueType),
+	string(queue.LevelQueueType),
+	string(queue.LevelUniqueQueueType),
+}
+
+func checkUniqueQueues(ctx context.Context, logger log.Logger, autofix bool) error {
+	for _, name := range uniqueQueueNames {
+		q := setting.GetQueueSettings(name)
+		if q.Type == "" {
+			q.Type = string(queue.PersistableChannelQueueType)
+		}
+		found := false
+		for _, typ := range levelqueueTypes {
+			if typ == q.Type {
+				found = true
+				break
+			}
+		}
+		if !found {
+			logger.Info("Queue: %s\nType: %s\nNo LevelDB", q.Name, q.Type)
+			continue
+		}
+
+		connection := q.ConnectionString
+		if connection == "" {
+			connection = q.DataDir
+		}
+
+		db, err := nosql.GetManager().GetLevelDB(connection)
+		if err != nil {
+			logger.Error("Queue: %s\nUnable to open DB connection %s: %v", q.Name, connection, err)
+			return err
+		}
+		defer db.Close()
+
+		prefix := q.Name
+
+		iQueue, err := levelqueue.NewQueue(db, []byte(prefix), false)
+		if err != nil {
+			logger.Error("Queue: %s\nUnable to open Queue component: %v", q.Name, err)
+			return err
+		}
+
+		iSet, err := levelqueue.NewSet(db, []byte(prefix+"-unique"), false)
+		if err != nil {
+			logger.Error("Queue: %s\nUnable to open Set component: %v", q.Name, err)
+			return err
+		}
+
+		qLen := iQueue.Len()
+		sMembers, err := iSet.Members()
+		if err != nil {
+			logger.Error("Queue: %s\nUnable to get members of Set component: %v", q.Name, err)
+			return err
+		}
+		sLen := len(sMembers)
+
+		if int(qLen) == sLen {
+			if qLen == 0 {
+				logger.Info("Queue: %s\nType: %s\nLevelDB: %s", q.Name, q.Type, "empty")
```

Comment:
```suggestion
				logger.Info("Queue: %s\nType: %s\nLevelDB: is empty", q.Name, q.Type)
```

Reply:
There's a reason why did this apparently odd thing... Take a look at the output.